Source/UINavigation/Private/UINavInputDisplay.cpp

```diff
@@ -98,9 +98,16 @@ void UUINavInputDisplay::UpdateInputVisuals()
 	if (IsValid(InputRichText)) InputRichText->SetVisibility(ESlateVisibility::Collapsed);
 	InputImage->SetVisibility(ESlateVisibility::Collapsed);
 
+	EInputRestriction Restriction = InputMethodRestriction;
+	if(Restriction == EInputRestriction::None)
+	{
+		// if set to None, detect automatically
+		Restriction = UINavPC->IsUsingGamepad() ? EInputRestriction::Gamepad : EInputRestriction::Keyboard_Mouse;
```

Comment:
In addition to this, only bind to the OnInputTypeChanged event in line 31 and 43 is the InputTypeRestriction is None.